impl Error

iamtakingithard opened the pull request:
@bltzxsd take a look
resolves https://github.com/AndreVuillemot160/QuickDash/issues/3

iamtakingithard committed:
Update error.rs

iamtakingithard committed:
Update error.rs

chronurgist commented on src/error.rs:
```diff
@@ -52,3 +56,16 @@ impl From<i32> for Error {
 }
+
+impl <'a> std::fmt::Display for Error <'_> {
+    //TODO: Add documentation 
```
No real need to add documentation here.

chronurgist commented on src/error.rs:
```diff
@@ -52,3 +54,16 @@ impl From<i32> for Error {
+    //TODO: Add documentation 
+    pub fn fmt(&self, f: &mut std::fmt::Formatter <'_> ) -> std::fmt::Result {
+        match *self {
+            Error::NoError(str: &str) => wirteln!(io::stderr(), "{}", str),
```
This needs to be removed. There is no need to handle a case where there is not an error since you return a concrete value anyway (empty tuple otherwise if no value needs to be returned but the operation can fail). IIRC this is also only used  in `main.rs`?

chronurgist commented on src/error.rs:
```diff
@@ -52,3 +54,16 @@ impl From<i32> for Error {
+    pub fn fmt(&self, f: &mut std::fmt::Formatter <'_> ) -> std::fmt::Result {
+        match *self {
+            Error::NoError(str: &str) => wirteln!(io::stderr(), "{}", str),
+            Error::OptionParsingError(str: &str) => wirteln!(io::stderr(), "{}", str),
```
Errors should should be using `write!` instead of `writeln!`

```suggestion 
            Error::OptionParsingError(err: &str) => write!(io::stderr(), "{}", err),
```

chronurgist commented on src/error.rs:
```diff
@@ -52,3 +54,16 @@ impl From<i32> for Error {
+        match *self {
+            Error::NoError(str: &str) => wirteln!(io::stderr(), "{}", str),
+            Error::OptionParsingError(str: &str) => wirteln!(io::stderr(), "{}", str),
+            Error::HashLengthDiffers(str: &str) => wirteln!(io::stderr(), "{}", str),
```
Same as above 

```suggestion 
            Error::HashLengthDiffers(err: &str) => write!(io::stderr(), "{}", err),
```

chronurgist commented on src/error.rs:
```diff
@@ -52,3 +54,16 @@ impl From<i32> for Error {
+            Error::NoError(str: &str) => wirteln!(io::stderr(), "{}", str),
+            Error::OptionParsingError(str: &str) => wirteln!(io::stderr(), "{}", str),
+            Error::HashLengthDiffers(str: &str) => wirteln!(io::stderr(), "{}", str),
+            Error::HashesFileParsingFailure(str: &str) => wirteln!(io::stderr(), "{}", str),
```
Same as above
```suggestion
            Error::HashesFileParsingFailure(err: &str) => write!(io::stderr(), "{}", err),
```

rawcptr reviewed (changes requested):
[std::error::Error](<https://doc.rust-lang.org/std/error/trait.Error.html>) impl needed!

iamtakingithard committed:
Update error.rs

Replace `(str: &str) => wirteln!(io::stderr(), "{}", str),` to `(err: &str) => write!(io::stderr(), "{}", err),`

iamtakingithard committed:
Update error.rs

Use solution provided by @bltzxsd 
If all checks are gonna be good, it will be merged

iamtakingithard committed:
Update lib.rs

iamtakingithard committed:
Update lib.rs

iamtakingithard committed:
Shut clippy

iamtakingithard committed:
Update lib.rs

iamtakingithard committed:
Update error.rs

chronurgist commented on src/error.rs:
```diff
@@ -29,12 +29,24 @@ pub enum Error<'err> {
+        }
     }
 }
 
```
alteast Remove the comments bro. Come on.

iamtakingithard committed:
Update rust.yml

iamtakingithard committed:
Merge branch 'main' into error-impl
